prow/plugins/lifecycle/lifecycle.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lifecycle
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const deprecatedWarn = true
+
+var (
+	deprecatedTick = time.Tick(time.Hour) // Warn once per hour
+	lifecycleRe    = regexp.MustCompile(`(?mi)^/(remove-)?(freeze|stale|putrid|rotten|frozen)( cancel)?\s*$`)
+)
+
+func init() {
+	plugins.RegisterGenericCommentHandler("lifecycle", lifecycleHandleGenericComment, nil)
+	logrus.SetLevel(logrus.DebugLevel)
+}
+
+type commentClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+}
+
+type lifecycleClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	AddLabel(owner, repo string, number int, label string) error
+	RemoveLabel(owner, repo string, number int, label string) error
+}
+
+func deprecate(gc commentClient, plugin, org, repo string, number int, e *github.GenericCommentEvent) error {
+	select {
+	case <-deprecatedTick:
```

Comment:
Once an hour until mid-2018 seems like a lot of comments!

Reply:
It only does this if your prow plugin has close and reopen enabled. If you instead enable lifecycle you'll get the functionality and no messages.

Reply:
@cjwagner how are we looking on tokens these days? do we have monitoring for this yet?

Reply:
Reread the whole thing, obviously this won't really matter for tokens unless you have an absurd number of repos using deprecated plugins.